store/root.js

```diff
@@ -85,7 +85,14 @@ class Root extends Store {
 
   @computed
   get dataProviders() {
-    return this.user.dataProviders
+    const ids = new Set()
+    return [this.dataProvider, ...this.user.dataProviders].filter(
+      (dataProvider) => {
+        if (dataProvider == null || ids.has(dataProvider.id)) return false
+        ids.add(dataProvider.id)
+        return true
+      }
```

Comment:
Why do you filter it?

Reply:
Due to implementation of the `changeDataProvider`, `this.dataProvider` can be `null`. I filter it to avoid an if-statements to check it with the following array unshifting.

Reply:
Ok that makes sense but what about the other filtering. Is there a possibility that you may have two data providers with the same id?

Reply:
This is what `ids.add(id)` stands for. If the set above already contains an ID, it's not added again. It's a trivial deduplication algorithm.

Reply:
Yep I understand the algorithm I was asking more why you added it there. Why are you worried about duplicate values?

Reply:
I assume that it may happen that the user's data providers are not loaded yet and the direct request was triggered, so there may be a duplicated repository